src/runner.ts

```diff
@@ -13,12 +14,12 @@ export const runRules = (
   let results: IRuleResult[] = [];
 
   for (const name in rules) {
-    if (!rules.hasOwnProperty(name)) continue;
+    if (!Object.prototype.hasOwnProperty.call(rules, name)) continue;
 
     const rule = rules[name];
     if (!rule) continue;
 
-    if (rule.hasOwnProperty('recommended') && !rule.recommended) {
+    if (rule.severity !== undefined && getDiagnosticSeverity(rule.severity) === -1) {
```

Comment:
Does `if (getDiagnosticSeverity(rule.severity) === -1) {` not work?

Reply:
Yeah, it does not expect undefined value to be provided. While it's unlikely rule.severity will be undefined at this very point as we normalize all rules beforehand, it's just safer in case rules haven't been processed by merger for some reason, i.e. they were added somehow manually using API. I suppose we could get rid of this once we drop addRules method as this is potentially the only way where a breakage could be introduced.